lib/noether/either.ex

```diff
@@ -21,6 +21,42 @@ defmodule Noether.Either do
   def map({:ok, a}, f), do: {:ok, f.(a)}
   def map(any = {:error, _}, _), do: any
 
+  @doc """
+  Given an `{:ok, value}` and a function, it flats the ok unwrapping and applying the function on the `value` returning `{:ok, f.(value)}`.
+  If an `{:error, _}` is given, it is returned as-is.
+
+  ## EXAMPLES
+    iex> flat_map({:ok, {:ok, -1}}, &Kernel.abs/1)
+    {:ok, 1}
+
+    iex> flat_map({:ok, -1}, &Kernel.abs/1)
+    ** (FunctionClauseError) no function clause matching in Noether.Either.flat_map/2
```

Comment:
This error could be specialized like the Enum protocol (Enumerable protocol not implemented for ..)